new Leetcode/49. Group Anagrams/49. Group Anagrams.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Created by sindhuhari on 1/19/18.
+ */
+var groupAnagrams = function (strs) {
+    if(!strs || strs.length === 0) {
+        return [];
+    }
+    var mapping = {};
+    var oldArray = strs.slice();
+    var sortStrings = strs;
+    var result = [];
+    for(var i = 0; i < sortStrings.length; i++) {
+        sortStrings[i] = sortString(sortStrings[i]);
+        if(mapping[sortStrings[i]]) {
+            mapping[sortStrings[i]].push(i)
+        } else {
+            mapping[sortStrings[i]] = [];
+            mapping[sortStrings[i]].push(i)
+        }
+    }
+    console.log(mapping)
+    for(var key in mapping) {
+        var group = [];
+        for(var value = 0 ; value < mapping[key].length; value++) {
+            var index = oldArray[mapping[key][value]];
```

Comment:
this variable is not index. it is the actual / original string. name is confusing.